-- Found rosidl_default_generators: 1.0.1 (/opt/ros/foxy/share/rosidl_default_generators/cmake)
-- Found ament_cmake_core: 0.9.9 (/opt/ros/foxy/share/ament_cmake_core/cmake)
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.8.10", minimum required is "3") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python3
-- Using all available rosidl_typesupport_c: rosidl_typesupport_fastrtps_c;rosidl_typesupport_introspection_c
-- Found rosidl_adapter: 1.2.1 (/opt/ros/foxy/share/rosidl_adapter/cmake)
-- Using all available rosidl_typesupport_cpp: rosidl_typesupport_fastrtps_cpp;rosidl_typesupport_introspection_cpp
-- Found PythonInterp: /usr/bin/python3 (found version "3.8.10") 
CMake Error at /opt/ros/foxy/share/rosidl_adapter/cmake/rosidl_adapt_interfaces.cmake:63 (message):
  execute_process(/usr/bin/python3 -m rosidl_adapter --package-name
  base_interfaces_demo --arguments-file
  /home/hw_learn/ws_ros2/ws01_plumbing/build/base_interfaces_demo/rosidl_adapter__arguments__base_interfaces_demo.json
  --output-dir
  /home/hw_learn/ws_ros2/ws01_plumbing/build/base_interfaces_demo/rosidl_adapter/base_interfaces_demo
  --output-file
  /home/hw_learn/ws_ros2/ws01_plumbing/build/base_interfaces_demo/rosidl_adapter/base_interfaces_demo.idls)
  returned error code 1:

  Error processing 'floate32 y' of 'base_interfaces_demo/Distance_Request':
  ''floate32' is an invalid message name.  It should have the pattern
  '^[A-Z][A-Za-z0-9]*$''

  Traceback (most recent call last):

    File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
      return _run_code(code, main_globals, None,
    File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
      exec(code, run_globals)
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/__main__.py", line 19, in <module>
      sys.exit(main())
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/main.py", line 53, in main
      abs_idl_file = convert_to_idl(
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/__init__.py", line 24, in convert_to_idl
      return convert_srv_to_idl(
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/srv/__init__.py", line 28, in convert_srv_to_idl
      srv = parse_service_string(package_name, input_file.stem, content)
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/parser.py", line 852, in parse_service_string
      request_message = parse_message_string(
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/parser.py", line 516, in parse_message_string
      Type(type_string, context_package_name=pkg_name),
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/parser.py", line 276, in __init__
      super(Type, self).__init__(
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/parser.py", line 204, in __init__
      raise InvalidResourceName(

  rosidl_adapter.parser.InvalidResourceName: 'floate32' is an invalid message
  name.  It should have the pattern '^[A-Z][A-Za-z0-9]*$'

Call Stack (most recent call first):
  /opt/ros/foxy/share/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake:130 (rosidl_adapt_interfaces)
  CMakeLists.txt:24 (rosidl_generate_interfaces)


-- Configuring incomplete, errors occurred!
See also "/home/hw_learn/ws_ros2/ws01_plumbing/build/base_interfaces_demo/CMakeFiles/CMakeOutput.log".
make: *** [Makefile:1228: cmake_check_build_system] Error 1
